src/mtype.d

```diff
@@ -8323,7 +8323,24 @@ public:
             {
                 return getProperty(e.loc, ident, flag);
             }
-            return sym.getMemtype(Loc()).dotExp(sc, e, ident, flag);
+
+            if (flag == 0)
+                flag = 1;  // silence lookup errors for memtype
+
+            Expression res = sym.getMemtype(Loc()).dotExp(sc, e, ident, flag);
```

Comment:
And. if the original `flag` == 1, following improved diagnostic messages should not be displayed.